src/components/sidebar/LineItems.tsx

```diff
@@ -28,14 +28,19 @@ class LineItems extends React.Component<ILineItemsProps> {
         }
     }
 
-    public filterLines = (lineNumber: string, transitType: TransitType) => {
-        const searchTargetAttributes = lineNumber;
+    public filterLines = (routes: ILineRoute[], lineId: string, transitType: TransitType) => {
+        let routeNames = '';
+        routes.forEach((lineRoute: ILineRoute) => {
+            routeNames += lineRoute.name;
+        });
+
+        const searchTargetAttributes = lineId + routeNames.toLowerCase();
         if (this.props.lineStore!.filters &&
             this.props.lineStore!.filters.indexOf(transitType) !== -1) {
             return false;
         }
 
-        if (searchTargetAttributes.includes(this.props.lineStore!.searchInput)) {
+        if (searchTargetAttributes.includes(this.props.lineStore!.searchInput.toLowerCase())) {
             return true;
         }
         return false;
```

Comment:
I would suggest replacing the search function with:

public filterLines = (routes: ILineRoute[], lineId: string, transitType: TransitType) => {
&nbsp;&nbsp;&nbsp;&nbsp;const searchTerm = this.props.lineStore!.searchInput.toLowerCase();
&nbsp;&nbsp;&nbsp;&nbsp;if (this.props.lineStore!.filters &&
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;this.props.lineStore!.filters.indexOf(transitType) !== -1)
&nbsp;&nbsp;&nbsp;&nbsp;{
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;return false;
&nbsp;&nbsp;&nbsp;&nbsp;} if (
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;routes
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.map(route => route.name.toLowerCase())
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;.find(name => name.indexOf(searchTerm) > -1)
&nbsp;&nbsp;&nbsp;&nbsp;) {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;return true;
&nbsp;&nbsp;&nbsp;&nbsp;} return false;
}

Since, adding the names using addition instead of handling them separately will cause incorrect search results.
Having the RoutePaths:
1 Tikkurila-Myyrmäki
2 Tikkurila-Myyrmäki
would cause positive search result when searching for the term 'kiti', even if the term never mentioned in the RoutePaths.